pycbc/waveform/generator.py

```diff
@@ -135,8 +137,13 @@ def dostuff(self):
     def _generate_from_current(self):
         """Generates a waveform from the current parameters.
         """
+        values_hash = hash(frozenset(**self.current_params.values()))
+        if values_hash == self._cache['hash']:
+            return self._cache['waveform']
         try:
-            return self.generator(**self.current_params)
+            waveform = self.generator(**self.current_params)
+            self._cache.update({'hash': values_hash, 'waveform': waveform})
+            return waveform
         except RuntimeError as e:
```

Comment:
This could be simplified...
```
try:
     return self._cache[values_hash]
except KeyError:
    try:
         waveform = self.generator(**self.current_params)
         self._cache.clear()
         self._cache[values_hash] = waveform
         return waveform
    except RunTimeError as e:
```